### ISSUE TITLE
A non-numeric value encountered in FrameReflower\Block.php in release 1.1.0

### ISSUE THREAD

**Issue by sakihl** [dompdf/dompdf#2645]
I am using PHP 7.4.16 and just upgraded to dompdf/dompdf 1.1.0. This did not happen in version 1.0.2.

Example html
````	<style>
		.container {
		width: 100%;
		padding-right: 15px;
		padding-left: 15px;
		margin-right: auto;
		margin-left: auto}
	</style>

	<div class="container">
		<p>Something Here</p>
	</div>
````
Gives the error "A non-numeric value encountered" in FrameReflower\Block.php at line 198, 
````
         if ($diff >= 0) {
                // Find auto properties and get them to take up the slack
                if ($width === "auto") {
                    $width = $diff;

                    if ($lm === "auto") {
                        $lm = 0;
                    }
                    if ($rm === "auto") {
                        $rm = 0;
                    }
                } elseif ($lm === "auto" && $rm === "auto") {
                    $lm = $rm = $diff / 2;
                } elseif ($lm === "auto") {
                    $lm = $diff;
                } elseif ($rm === "auto") {
                    $rm = $diff;
                }
            } else {
                // We are over constrained--set margin-right to the difference
                $rm = $rm + $diff;  <--- This is line 198

````
In this case `$diff < 0` and `$rm` is `"auto"` so `$rm = $rm + $diff` fails.

In other issues such as [2347](https://github.com/dompdf/dompdf/issues/2347) invalid html structure has been blamed. However here the code for `$diff >=0` clearly treats `$rm === "auto"` as a valid value so it seems as if the `else` code should handle it also.

**Comment by Mellthas** [comment]
Thanks for the catch! This is indeed an oversight, fixed by #2646.

**Comment by deefour** [comment]
@Mellthas can you tag a new patch release for this? It's breaking our builds.

**Comment by Mellthas** [comment]
Update is coming soon. In the meantime, you could temporarily set the version to the specific commit a18630f3f22b97cbf78eec773c44d49b7ac311a5 if you are using Composer, or otherwise apply the fix locally.

**Comment by Mellthas** [comment]
By the way, you might want to check whether you actually want `width: 100%` on the container, as it will be larger than its parent with the additional padding. A width value of `100%` was wrongly treated as `auto` in previous releases.

**Comment by deefour** [comment]
@Mellthas I'm locking to that commit locally to prove tests pass, but adding a Github OAuth token to our CI/CD pipeline for this temporary issue isn't worth it.

In the meantime, **huge** thanks for pointing out the width: 100%. I'm going to experiment with this tomorrow. We have a single instance of

    <table style="width:100%">

in the template that's breaking. I'm going to try adjusting this.

**Comment by bsweeney** [comment]
Just released the 1.1.1 version.

**Comment by deefour** [comment]
@bsweeney @Mellthas thank you so much for this.
